returnsrcvr-checker/src/main/java/org/checkerframework/checker/framework/AutoValueSupport.java

```diff
@@ -0,0 +1,48 @@
+package org.checkerframework.checker.framework;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+import org.checkerframework.checker.returnsrcvr.ReturnsRcvrAnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.TypesUtils;
+
+import com.google.auto.value.AutoValue;
+
+/**
+ * 
+ * AutoValue support for returns receiver checker, check if the current class or the super class is AutoValue Builder 
```

Comment:
Grammar: "AutoValue support for returns receiver checker. Checks if the current class or its super class is an AutoValue Builder"